.github/workflows/test.yml

```diff
@@ -0,0 +1,92 @@
+name: Test and build
+on:
+  pull_request:
+    types: [opened]
+  push:
+jobs:
+  tests:
+    name: Test with go version ${{ matrix.go_version }} on OS ${{matrix.os}} 
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        go_version: [1.19]
+        os: [ubuntu-latest]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use golang ${{ matrix.go_version }}
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go_version }}
+      - name: Go version
+        run: go version
+      - name: Go get dependencies
+        run: go get -v -t -d ./...
+      - name: Run tests
+        run: make test
+      - name: Send the coverage output
+        uses: shogo82148/actions-goveralls@v1
+        with:
+          path-to-profile: coverage.out
+
+  build:
+    name: Build docker image
+    needs: tests
+    runs-on: ubuntu-latest
+    if: ${{ startsWith(github.ref, 'refs/tags/') || github.ref == format('refs/heads/{0}', github.event.repository.default_branch) }}
+    
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        
+      - name: Configure docker metadata
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ghcr.io/mia-platform/config-sheperd
+            docker.io/miaplatform/config-sheperd
+          tags: |
+            type=ref,event=branch
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
+          labels: |
+            org.opencontainers.image.documentation=https://github.com/mia-platform/config-shepherd/blob/main/README.md
+            org.opencontainers.image.vendor=Mia Platform
+            
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Docker Login to ghcr.io
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Docker Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.BOT_DOCKER_USERNAME }}
```

Comment:
@davidebianchi can you check if the name of the variable is correct?

Reply:
Yes, they are correct